packages/runtime/src/runtimes/react/controls/rich-text.tsx

```diff
@@ -0,0 +1,161 @@
+   * possible.
+   *
+   * The approach we take here is to commit the prop data at an opportune time: as the user is
+   * typing we avoid to commit prop data. But once they've stopped typing, we commit it as soon as
+   * possible. This is known as a debounce.
+   */
+  const [value, setValue] = useState(() => {
+    const { selection, ...textWithoutSelection } = data ?? descriptor.config.defaultValue
+
+    return Value.fromJSON(textWithoutSelection)
+  })
+  const [shouldCommit, setShouldCommit] = useState(true)
+
+  useEffect(() => {
+    if (shouldCommit) {
+      const nextValue = Value.fromJSON(data ?? descriptor.config.defaultValue)
+
+      setValue(currentValue =>
+        currentValue.selection.isBlurred
+          ? Value.fromJSON(nextValue.toJSON({ preserveSelection: false }))
+          : nextValue,
+      )
+    }
+  }, [shouldCommit, data])
+
+  useEffect(() => {
+    if (shouldCommit) return
+
+    const timeoutId = window.setTimeout(() => {
+      setShouldCommit(true)
+    }, COMMIT_DEBOUNCE_DELAY)
+
+    return () => {
+      window.clearTimeout(timeoutId)
+    }
+  }, [shouldCommit])
+
+  function handleChange(change: OnChangeParam) {
+    setValue(change.value as Value)
+
+    if (change.value !== value) {
+      setShouldCommit(false)
+
+      control?.onChange(change)
+    }
+  }
+
+  const lastControl = useRef(control)
+  if (lastControl.current !== control) lastControl.current = control
+  const handleFocus = useCallback(() => {
+    lastControl.current?.focus()
+  }, [])
+  const handleKeyDown = useCallback(
+    (event: ReactKeyboardEvent, _editor: Editor, next: () => any) => {
+      if (Hotkeys.isUndo(event)) {
+        lastControl.current?.undo()
+
+        return true
+      }
+
+      if (Hotkeys.isRedo(event)) {
+        lastControl.current?.redo()
+
+        return true
+      }
+
+      if (isHotkey('escape')(event)) {
+        lastControl.current?.blur()
+
+        return true
+      }
+
+      return next()
+    },
+    [],
+  )
+
+  const handleBlur = useCallback((event: ReactFocusEvent, editor: Editor, next: () => any) => {
+    const selection = editor.value.selection
+
+    next()
+
+    // Normally, after a user highlight a text, clicking on the panel will remove the text selection.
+    // This line is a workaround for that. Because the panel is not in the iframe, relatedTarget
+    // would be null, and we select the previous text selection to maintain the text highlight.
+    // Inspiration: https://github.com/ianstormtaylor/slate/issues/3412#issuecomment-663906003
+    if (event.relatedTarget == null) editor.select(selection)
+  }, [])
+
+  const isInBuilder = useIsInBuilder()
+  return (
+    <RichTextEditor
+      ref={setEditor}
+      readOnly={!isInBuilder || control == null}
+      value={value}
+      onChange={handleChange}
+      onFocus={handleFocus}
+      onKeyDown={handleKeyDown}
+      onBlur={handleBlur}
+    />
```

Comment:
This doesn't seem to be leveraging the optimizations I shipped for v0.6 for live pages. Given the similarity between this and our built-in Text component, is there a reason you didn't use that component directly?